### PR TITLE
docs: add configuration documentation

### PR DESCRIPTION
**Pull request by technicallyty**
## Overview

adds some simple documentation about our configuration values.

## Brief Changelog

config config config. i also fixed a typo.

## Testing and Verifying

run `make world-docs` to and navigate to http://localhost:3000/cardinal/game/configuration

### CODE REVIEW

**Comment by technicallyty** [comment]
seems the markdown lint job is faulty.

File: docs/cardinal/game/configuration.mdx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+Below are the key value pairs that can be set in either the `world.toml` file or directly in environment variables.
+
+
+## Cardinal `[cardinal]`
+| Name                         | Description                                                                                                                                                                                                          | Required for Production |
+|------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------|
+| CARDINAL_NAMESPACE           | The name of your cardinal game shard. If you are running multiple game shards in a single World Engine instance, this MUST be unique for each game shard. Must only be alphanumeric characters.                      | YES                     |
```
Comment (jerargus) [inline docs/cardinal/game/configuration.mdx:14]: Nit: The namespace may also contain dashes.

File: docs/cardinal/game/configuration.mdx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+|-----------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| DA_AUTH_TOKEN   | The authentication token required to communicate with the Celestia DA Layer. This is typically generated by the celestia light client, and can be found in the logs during startup                    |
+| DA_BASE_URL     | The base URL of the DA layer.                                                                                                                                                                         |
+| DA_NAMESPACE_ID | The namespace the EVM will post its blobs under. Learn more about DA namespaces here [Namespaces](https://celestia.org/glossary/namespaced-merkle-tree/) Can be generated with `openssl rand -hex 10` |                                                                                                                 |
```
Comment (jerargus) [inline docs/cardinal/game/configuration.mdx:26]: There's missing punctuation before "Can be generated...".

File: docs/cardinal/game/configuration.mdx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+| KEY_MNEMONIC    | The mnemonic of the account to instantiate the chain with. This will allocate some funds to the account.                                                                                              |
+| FAUCET_ADDR     | Supplying a faucet address enables the given account a supply of funds to use for faucet purposes.                                                                                                    |
+| BLOCK_TIME      | The interval at which the EVM Base Shard should produce blocks. Examples: `1s`, `2s`, `3s`...                                                                                                         |
+
+## Nakama Relayer `[nakama]`
+| Name                | Description                                                                                                                                 |
+|---------------------|---------------------------------------------------------------------------------------------------------------------------------------------|
+| ENABLE_ALLOWLIST    | Can be either **true** or **false**. Enables the allowlist functionality of Nakama. This gives Nakama endpoints to generate and claim keys. |
```
Comment (jerargus) [inline docs/cardinal/game/configuration.mdx:35]: We should have a section in our docs describing how to use the allow list feature. I made a ticket: https://linear.app/arguslabs/issue/WORLD-1052/document-allow-list-feature

**Review by jerargus — APPROVED** [review]

File: docs/cardinal/game/configuration.mdx (hunk 1 above)
Comment (technicallyty) [reply docs/cardinal/game/configuration.mdx]: fixed [nits](https://github.com/Argus-Labs/world-engine/pull/736/commits/4e49fffad477551015c40df5da722fea99730694)

File: docs/cardinal/game/configuration.mdx (hunk 2 above)
Comment (technicallyty) [reply docs/cardinal/game/configuration.mdx]: fixed [nits](https://github.com/Argus-Labs/world-engine/pull/736/commits/4e49fffad477551015c40df5da722fea99730694)

File: docs/cardinal/game/configuration.mdx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+|------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------|
+| CARDINAL_NAMESPACE           | The name of your cardinal game shard. If you are running multiple game shards in a single World Engine instance, this MUST be unique for each game shard. Must only be alphanumeric characters and hyphens (-).      | YES                     |
+| CARDINAL_MODE                | The cardinal mode determines how your game shard runs. Must be either "development" or "production". In development, Cardinal will not submit any transactions to the base shard, and security concerns are lowered. | N/A                     |
+| REDIS_ADDRESS                | The address of the redis instance.                                                                                                                                                                                   | YES                     |
```
Comment (smsunarto) [inline docs/cardinal/game/configuration.mdx:16]: nit: capitalized Redis

File: docs/cardinal/game/configuration.mdx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+| REDIS_PASSWORD               | The password of your redis instance.                                                                                                                                                                                 | YES                     |
+| CARDINAL_LOG_LEVEL           | The level of logging Cardinal should output. See [Leveled Logging](https://github.com/rs/zerolog?tab=readme-ov-file#leveled-logging) for more details.                                                               | NO                      |
```
Comment (smsunarto) [inline docs/cardinal/game/configuration.mdx:18]: Provide list of valid log level strings

File: docs/cardinal/game/configuration.mdx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+Below are the key value pairs that can be set in either the `world.toml` file or directly in environment variables.
+
+
+## Cardinal `[cardinal]`
+| Name                         | Description                                                                                                                                                                                                          | Required for Production |
+|------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------|
+| CARDINAL_NAMESPACE           | The name of your cardinal game shard. If you are running multiple game shards in a single World Engine instance, this MUST be unique for each game shard. Must only be alphanumeric characters and hyphens (-).      | YES                     |
```
Comment (smsunarto) [inline docs/cardinal/game/configuration.mdx:14]: Cardinal should be capitalized.

brand/product names should always be capitalized (i.e. World Engine, Redis, Cardinal, Nakama, etc)

**Review by smsunarto — DISMISSED** [review]
i left some comments, but I'd prefer if we hold back on merging this because a lot of this are not up to date with my config PR.

File: docs/cardinal/game/configuration.mdx (hunk 6 above)
Comment (technicallyty) [reply docs/cardinal/game/configuration.mdx]: fixed [nits](https://github.com/Argus-Labs/world-engine/pull/736/commits/ec1345c83691c5e1ea44e62550f72af71b52e2d1)

File: docs/cardinal/game/configuration.mdx (hunk 5 above)
Comment (technicallyty) [reply docs/cardinal/game/configuration.mdx]: fixed [nits](https://github.com/Argus-Labs/world-engine/pull/736/commits/ec1345c83691c5e1ea44e62550f72af71b52e2d1)

File: docs/cardinal/game/configuration.mdx (hunk 4 above)
Comment (technicallyty) [reply docs/cardinal/game/configuration.mdx]: fixed [nits](https://github.com/Argus-Labs/world-engine/pull/736/commits/ec1345c83691c5e1ea44e62550f72af71b52e2d1)

**Comment by technicallyty** [comment]
> i left some comments, but I'd prefer if we hold back on merging this because a lot of this are not up to date with my config PR.

okay will hold til that is merged

**Review by jerargus — APPROVED** [review]
Looks fine, however it's missing the "BASE_SHARD_ROUTER_KEY", and "CARDINAL_ROLLUP_ENABLED" config as well as the TELEMETRY_* configs (TELEMETRY_ENABLED, TELEMETRY_STATSD_ADDRESS, and TELEMETRY_TRACE_ADDRESS).

**Comment by technicallyty** [comment]
> Looks fine, however it's missing the "BASE_SHARD_ROUTER_KEY", and "CARDINAL_ROLLUP_ENABLED" config as well as the TELEMETRY_* configs (TELEMETRY_ENABLED, TELEMETRY_STATSD_ADDRESS, and TELEMETRY_TRACE_ADDRESS).

updated

File: docs/cardinal/game/configuration.mdx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+
+## Cardinal `[cardinal]`
+| Name                         | Description                                                                                                                                                                                                                          | Required in Rollup Mode |
+|------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------|
+| CARDINAL_NAMESPACE           | The name of your Cardinal game shard. If you are running multiple game shards in a single World Engine instance, this MUST be unique for each game shard. Must only be alphanumeric characters and hyphens (-).                      | YES                     |
+| ROLLUP_MODE                  | When false, Cardinal will behave like a traditional game server. When true, Cardinal will use the EVM base shard to sequence game transactions, as well as enable the Router.                                                        | N/A                     |
```
Comment (jerargus) [inline docs/cardinal/game/configuration.mdx:15]: This should be "CARDINAL_ROLLUP_ENABLED". The description can remain the same.

https://github.com/Argus-Labs/world-engine/blob/main/cardinal/config.go#L53

Comment (technicallyty) [reply docs/cardinal/game/configuration.mdx]: thanks, fixed.

**Pull request closed without merging**